src/app/errors.rs

```diff
@@ -126,6 +127,11 @@ fn map_err_to_5xx(e: Error) -> StatusCode {
     } else if let Some(_) = e.downcast_ref::<router::NotRecognized>() {
         error!("could not recognize request");
         http::StatusCode::BAD_GATEWAY
+    } else if let Some(err) =
+        e.downcast_ref::<outbound::require_identity_on_endpoint::RequireIdentityError>()
+    {
+        error!("{}", err);
+        http::StatusCode::FORBIDDEN
```

Comment:
Actually, I know I said this should be FORBIDDEN in the issue, but that seems to imply something else -- that the server forbade requests from this client, when, in reality, the client's requirement on the server was not met. PRECONDITION FAILED seems inappropriate as well, though perhaps more in the direction of what we're looking for.

@seanmonstar any suggestions?

Reply:
Hm good point... FORBIDDEN should be something returned by the server and that's not the case here. I can wait for input from @seanmonstar, but I like the suggestion.

Reply:
how about [`409 Conflict`](https://tools.ietf.org/html/rfc7231#section-6.5.8)

Reply:
"428 Precondition Failed" is meant for when using headers like `If-Match: some-etag` in a `PUT`, `POST`, etc.

I've been thinking we should be able to send HTTP2 reset codes in here if the request was HTTP2, and `INADEQUATE_SECURITY` seems to match...

Reply:
https://github.com/linkerd/linkerd2/issues/3111